src/pages/Homepage/Homepage.vue

```diff
@@ -85,13 +93,33 @@ export default {
 			return client.query({
 				query: homePageQuery
 			}).then(() => {
-				return ContentfulPage();
-			}).then(resolvedImport => {
+				return Promise.all([
+					client.query({ query: experimentAssignmentQuery, variables: { id: 'new_home_layout' } }),
+					ContentfulPage(),
+				]);
+			}).then(result => {
 				// Call preFetch for the active homepage
-				const component = resolvedImport.default;
-				return preFetchAll([component], client, args);
+				const expVersion = result[0]?.data?.experiment?.version;
+				if (expVersion === 'a' || expVersion === 'b') {
```

Comment:
I think this is all the versions? you could simplify to just `if (expVersion)`

Reply:
We actually need to preserve a version `c` which represents a percentage of traffic we'll send to cms-page-server.

Still a concerned that it's set as the default with no controls to bypass. While we could use the exp distribution settings it would still be better to have an explicit `if (expVersion === 'c')` block to activate that redirect.

Reply:
Ok, I'll add the validation!